src/factories/web3/Web3Modal.ts

```diff
@@ -69,6 +80,11 @@ export default {
     this.provider.on("accountsChanged", accountsChanged || NOOP);
     this.provider.on("chainChanged", chainChanged || NOOP);
     this.provider.on("connect", connect || NOOP);
-    this.provider.on("accountsChanged", disconnect || NOOP);
+    this.provider.on("disconnect", disconnect || NOOP);
```

Comment:
@whatl3y I think this bind was referencing the wrong callback `accountsChanged` instead of `disconnect`.